### PR TITLE
Adds expansive grounds around the refugee center

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
```SUMMARY: Content "Adds expansive grounds around the refugee center"```

#### Purpose of change
The refugee center is frequently wiped out by ants and fungaloids, and occasionally triffids.
As the primary point of contact with anything resembling a plot, it would be nice if it was a little more resilient.

In addition, I would like to be able to use update_mapgen to add stuff to the surrounding area, allowing caravans to set up there or buildings to be built.

#### Describe the solution
Changes the refugee center to a 15x15 special, by adding eight new 5x5 maps all around the center. These are named with a simple coordinate system so that update_mapgen orders can easily figure out which tile to change. This should help a small amount to keep enemy spawns away from the center, as a buffer.

Adds a broken fence around the perimeter. Not much good for keeping zombies out, as it's in a bad state of disrepair.

~~Adds (when draft done) a parking lot where caravans gather.

Also (when draft done) adds 2 mapped z-levels above and 4 below the center. These are empty maps, but placing them will allow me to add stuff to them without re-creating the center later. It's possible I'll add  the center roof in this PR, I do need it.~~

I will do these additions in a follow up PR or two.

#### Describe alternatives you've considered
I have considered making the map 25x25. If I do that, the fence will remain in its current location. Currently I'd like to see how much this buffer reduces incoming threats to the center.

#### Additional context
Whole lotta this:
image

Little bitta this:
image

Looks like this on the overmap:
image

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
Off-topic: I always wondered what type of building this refugee center was prior to the Cataclysm?

**Comment by SunshineDistillery** [comment]
You should try the new predecessor_terrain for the mostly empty lawn.  Use fields, and then have the map just place the fences.  Look at the dock om_terrain for an example.

**Comment by nexusmrsep** [comment]
Other possibility would be a perimeter of traps, spike pits etc. 
It could also be an opportunity for a quest "Help fortify the center" or similar.

**Comment by I-am-Erk** [comment]
I've now set up the things this needs. Still a WIP but I'm taking it off draft because I'm getting a weird error I hope travis will clarify for me.

**Comment by I-am-Erk** [comment]
@nexusmrsep missions to fortify the center are one of my main reasons for doing this. Now that the overmaps around the center all have unique id's I can use update_mapgen on them.

@Night-Pryanik I am going with the idea that the center was purpose build like the evacuation shelters. I think it probably was an old military site that got repurposed.

**Comment by Amoebka** [comment]
Perhaps the spawning rate of refugee centers should go down soon? As the center gets expanded more and more, it feels more and more weird to encounter character clones when you travel to new overmaps. Having most centers being overrun by ants/fungus/triffids **intentionally** might actually alleviate the issue somewhat.

**Comment by Mecares** [comment]
Randomised names would also help

**Comment by I-am-Erk** [comment]
At some point, the free Merchant center will be unique and every other refugee center will be overrun and dead, or at most have a few disorganized survivors in it.

For now, since NPC death can't be scripted for currently and screws up missions, the duplication is a feature that allows you to start again if someone dies, or if the game updates and you want to play the new missions.

Giving the refugees random names would make it impossible for them to talk about each other.

**Comment by I-am-Erk** [comment]
This is ready now. I have decided not to include any content in this besides the map itself, because at 20,000 lines, anything clever I slip in will be impossible to review. I will make a separate PR or PRs that adds some details and features.

**Comment by kevingranade** [comment]
Ok sanity check here, do we REALLY need to check in 10,000+ lines of mostly redundant open space mapgen in order to meet this goal? Yes they need distinct names for later static upgrade placement, but we should be able to at least alias these things to each other instead of repeating them over and over.

**Comment by SunshineDistillery** [comment]
If you want to shrink the size, you really don't need to define rows at all.  Make or use predecessors for fields and lots.  Then make six chunks for fences, cardinal directions and corners.  Then for each individual om_terrain, just place the appropriate nested chunk.  Probably the gate would need a normal om_terrain.

**Comment by SunshineDistillery** [comment]
Off topic: 
I thought of an easy way to make the FM evac shelter a unique location.
1. Change current evac shelter special to 0 frequency spawn.
2. Make a new abandoned version of the evac shelter to take its place.
3. Use the computer mission to find the (original) shelter, 
4. When it can't find it, it will then place the shelter at the nearest legal position.

It wouldn't be truly unique, because you could spawn more from other start shelters, unless you changed the comp mission to use variables. Probably better ways to do it, but that's a fast mostly json answer.
Obviously, that would be another PR.

**Comment by I-am-Erk** [comment]
I condensed 9000 lines and removed a few dozen files.

The remaining lines are the surface level fields, which are pretty boring at the moment but I plan to add some unique buildings to. While we could remove them, a lot of them are going to get filled back up and I'm not sure how much it adds to further condense it. I can though if you want. 

I'm not really super keen on using predecessor to make this out of fields. I could, but these aren't really fields, they're areas that were landscaped partly and then left to seed when the project was dropped.

File: data/json/overmap_terrain_evac_center.json [hunk 1]
```diff
@@ -204,5 +203,7779 @@
...
     "sym": "┘",
     "color": "dark_gray"
+  },
+  {
+    "type": "overmap_terrain",
+    "abstract": "generic_rc_grounds",
+    "see_cost": 5,
+    "name": "field",
+    "sym": ".",
+    "color": "green"
```
Comment (kevingranade) [inline data/json/overmap_terrain_evac_center.json]:
```suggestion
    "color": "brown"
```

**Comment by kevingranade** [comment]
Road connection got messed up
2019-06-14-075013_1366x768_scrot
2019-06-14-075025_1366x768_scrot

**Comment by I-am-Erk** [comment]
Looks like I forgot to update a file from the playtest version, has both roads and colour change. Will get that in a minute.

**Comment by AngularFruit** [comment]
> Off-topic: I always wondered what type of building this refugee center was prior to the Cataclysm?

Off-Topic: 
50bfc775709bb image
I've always imagined it like an old National Guard Armory, like the one in my town.

**Comment by AngularFruit** [comment]
> Off-topic: I always wondered what type of building this refugee center was prior to the Cataclysm?

50bfc775709bb image
Off-topic: I've always thought if it like as an old National Guard Armory, like the one in my town.

**Comment by I-am-Erk** [comment]
I picture much the same thing actually. When there's a chance to add an NPC who knows about it I was planning on having it be an old armoury and decommissioned cold war bunker that were repurposed when people wanted evacuation preparedness in the late pre-cataclysm days

**Pull request merged**